src/Tribe/Template/Month.php

```diff
@@ -314,13 +319,41 @@ protected function produce_json_ld_markup( $echo = false ) {
 			}
 
 			$events = wp_list_pluck( $this->events_in_month, 'ID' );
+			if ( tribe_is_event_category() ) {
+				$events = $this->filter_by_current_term( $events );
+			}
+
 			Tribe__Events__JSON_LD__Event::instance()->markup( $events );
 
 			if ( ! $echo ) {
 				return ob_get_clean();
 			}
 		}
 
+		/**
+		 * Remove any event that is not the same as the current term of the array of events, it return a modified events
+		 * array with the events that only has the current term or original events if the term is not valid.
+		 *
```

Comment:
Add a `@since` param to the method, even if it's a Private one.

Reply:
Added.